Source/BoundaryConditions/PML.cpp

```diff
@@ -759,9 +765,58 @@ PML::PML (const int lev, const BoxArray& grid_ba, const DistributionMapping& /*g
 }
 
 BoxArray
-PML::MakeBoxArray (const amrex::Geometry& geom, const amrex::BoxArray& grid_ba,
+PML::MakeBoxArray (const amrex::Box& regular_domain,
+                   const amrex::Geometry& geom, const amrex::BoxArray& grid_ba,
                    int ncell, int do_pml_in_domain,
                    const amrex::IntVect do_pml_Lo, const amrex::IntVect do_pml_Hi)
+{
+    if (regular_domain.numPts() == grid_ba.numPts()) {
+        return MakeBoxArray_single(regular_domain, grid_ba, ncell, do_pml_Lo, do_pml_Hi);
+    } else {
```

Comment:
Would it be possible to add some comment along the lines of the following one (I hope it's correct to refer to `grid_ba` as the "PML grids" ...)?
```suggestion
    } else { // the union of the PML grids is *not* a single rectangular domain
```